ui/app/services/control-group.js

```diff
@@ -0,0 +1,56 @@
+import Ember from 'ember';
+import ControlGroupError from 'vault/lib/control-group-error';
+
+const { Service, assign, inject, RSVP } = Ember;
+
+// list of endpoints that return wrapped responses
+// without `wrap-ttl`
+const WRAPPED_RESPONSE_PATHS = [
+  'sys/wrapping/rewrap',
+  'sys/wrapping/wrap',
+  'sys/replication/performance/primary/secondary-token',
+  'sys/replication/dr/primary/secondary-token',
+];
+
+export default Service.extend({
+  version: inject.service(),
+  router: inject.service(),
+
+  checkForControlGroup(callbackArgs, response, wasWrapTTLRequested) {
+    let creationPath = response && Ember.get(response, 'wrap_info.creation_path');
+    if (
+      this.get('version.isOSS') ||
+      wasWrapTTLRequested ||
+      !response ||
+      (creationPath && WRAPPED_RESPONSE_PATHS.includes(creationPath)) ||
+      (response && !response.wrap_info)
```

Comment:
Since you're returning if `!response` is true above, I think you can simplify this like to `!response.wrap_info`.

Reply:
Oh yep, will change - also talked to the backend devs and this whole thing will be getting much simpler, just checking one flag once its in place.